sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -349,9 +349,10 @@ export class EventHubSender extends LinkEntity {
       log.sender("[%s] Sender '%s', trying to send EventData[].", this._context.connectionId, this.name);
       const messages: AmqpMessage[] = [];
       // Convert EventData to AmqpMessage.
-      for (let i = 0; i < data.length; i++) {
-        const message = EventDataInternal.toAmqpMessage(data[i]);
-        message.body = this._context.dataTransformer.encode(data[i].body);
+      for (let i = 0; i < events.length; i++) {
+        const batchLabel = options && options.batchLabel!;
```

Comment:
This constant is getting created for every iteration of the for loop, I'd suggest to pull it out of the loop, as the value doesnt change between iterations.

Also, what is the reason for adding `!` here? That is used to tell the typescript that even though the type allows `undefined`, we are sure that it is not `undefined`.

In this case, we are not sure that the batchLabel is not undefined